packages/typespec-client-generator-core/test/types.test.ts

```diff
@@ -930,7 +930,37 @@ describe("typespec-client-generator-core: types", () => {
       await helper("@azure-tools/typespec-csharp", "Enum1", "One");
       await helper("@azure-tools/typespec-java", "JavaEnum1", "JavaOne");
     });
+
+    it("union as enum rename", async () => {
+      const { TestUnion } = (await runner.compileWithCustomization(
+        `
+        @service({})
+        namespace N {
+          @test
+          union TestUnion{
+            @clientName("ARename")
+            "A",
+            "B": "B_v",
+            string
+          }
+          op x(body: TestUnion): void;
+        }
+      `,
+        `
+        namespace Customizations;
+
+        @@clientName(N.TestUnion, "TestUnionRename");
+        @@clientName(N.TestUnion.B, "BRename");
```

Comment:
There is one issue right in that you cannot augment `A`?

This doesn't make this change wrong but I think it greatly limit its purpose

Reply:
Yeah. I suppose it is by design since union variant `A` does not has an explicit name.